### PR TITLE
Added new GitHub issue form for React DevTools bug reports

### PR DESCRIPTION
**Pull request by bvaughn**
- [x] Add a new GitHub issue [form template](https://gh-community.github.io/issue-template-feedback/structured/) for DevTools bug reports.
- [x] Update the React DevTools issue reporting flow to 

You can demo this new form in action [here](https://github.com/bvaughn/test-github-issue-template/issues/new?template=devtools_bug_report.yml&labels=Component:%20Developer%20Tools,Type:%20Bug,Status:%20Unconfirmed&title=%5BDevTools%20Error%5D%20This%20is%20an%20example%20error&automated_package=react-devtools-inline&automated_version=4.13.1-bd070eb2c4&automated_error_message=This%20is%20an%20example%20error&automated_call_stack=%20%20%20%20at%20Components%20(webpack-internal:///../react-devtools-shared/src/devtools/views/Components/Components.js:43:9)%0A%20%20%20%20at%20renderWithHooks%20(webpack-internal:///../../build/node_modules/react-dom/cjs/react-dom.development.js:2606:157)%0A%20%20%20%20at%20mountIndeterminateComponent%20(webpack-internal:///../../build/node_modules/react-dom/cjs/react-dom.development.js:3035:886)%0A%20%20%20%20at%20beginWork%20(webpack-internal:///../../build/node_modules/react-dom/cjs/react-dom.development.js:3356:93)%0A%20%20%20%20at%20beginWork$1%20(webpack-internal:///../../build/node_modules/react-dom/cjs/react-dom.development.js:4320:93)%0A%20%20%20%20at%20performUnitOfWork%20(webpack-internal:///../../build/node_modules/react-dom/cjs/react-dom.development.js:4168:270)%0A%20%20%20%20at%20workLoopSync%20(webpack-internal:///../../build/node_modules/react-dom/cjs/react-dom.development.js:4154:30)%0A%20%20%20%20at%20renderRootSync%20(webpack-internal:///../../build/node_modules/react-dom/cjs/react-dom.development.js:4150:108)%0A%20%20%20%20at%20performConcurrentWorkOnRoot%20(webpack-internal:///../../build/node_modules/react-dom/cjs/react-dom.development.js:4029:126)%0A%20%20%20%20at%20workLoop%20(webpack-internal:///../../build/node_modules/scheduler/cjs/scheduler.development.js:269:38)&automated_component_stack=%0A%20%20%20%20at%20Components%20(webpack-internal:///../react-devtools-shared/src/devtools/views/Components/Components.js:43:9)%0A%20%20%20%20at%20ErrorBoundary%20(webpack-internal:///../react-devtools-shared/src/devtools/views/ErrorBoundary/ErrorBoundary.js:35:5)%0A%20%20%20%20at%20PortaledContent%20(webpack-internal:///../react-devtools-shared/src/devtools/views/portaledContent.js:28:5)%0A%20%20%20%20at%20div%0A%20%20%20%20at%20div%0A%20%20%20%20at%20ProfilerContextController%20(webpack-internal:///../react-devtools-shared/src/devtools/views/Profiler/ProfilerContext.js:31:3)%0A%20%20%20%20at%20TreeContextController%20(webpack-internal:///../react-devtools-shared/src/devtools/views/Components/TreeContext.js:664:3)%0A%20%20%20%20at%20SettingsContextController%20(webpack-internal:///../react-devtools-shared/src/devtools/views/Settings/SettingsContext.js:30:3)%0A%20%20%20%20at%20ModalDialogContextController%20(webpack-internal:///../react-devtools-shared/src/devtools/views/ModalDialog.js:52:3)%0A%20%20%20%20at%20DevTools%20(webpack-internal:///../react-devtools-shared/src/devtools/views/DevTools.js:75:3)%0A%20%20%20%20at%20DevTools&automated_github_query_string=https://api.github.com/search/issues?q=This%20is%20an%20example%20error%20in%3Atitle%20is%3Aissue%20is%3Aopen%20is%3Apublic%20label%3A%22Component%3A%20Developer%20Tools%22%20repo%3Afacebook%2Freact).

### Demo
https://user-images.githubusercontent.com/29597/117450144-dac9e700-af0e-11eb-8b91-07401b21f42b.mp4

### CODE REVIEW

File: packages/react-devtools-shared/src/devtools/views/ErrorBoundary/ReportNewIssue.js [hunk 1]
```diff
@@ -13,9 +13,18 @@ import {searchGitHubIssuesURL} from './githubAPI';
 import styles from './shared.css';
 
 function encodeURIWrapper(string: string): string {
-  return encodeURI(string).replace(/#/g, '%23');
```
Comment (eps1lon) [inline packages/react-devtools-shared/src/devtools/views/ErrorBoundary/ReportNewIssue.js:16]: Any reason you didn't use `encodeURIComponent(string)` instead which has `replace(/#/g, '%23')` built in?

File: packages/react-devtools-shared/src/devtools/views/ErrorBoundary/ReportNewIssue.js [hunk 2]
```diff
@@ -13,9 +13,18 @@ import {searchGitHubIssuesURL} from './githubAPI';
...
+  return encodeURI(string).replace(/\n/g, '%0A');
```
Comment (eps1lon) [inline packages/react-devtools-shared/src/devtools/views/ErrorBoundary/ReportNewIssue.js:16]: Is this a browser quirk? In chrome 90 both `encodeURI` and `encodeURIComponent` already replace `\n`.
Screenshot from 2021-05-07 10-43-04

File: packages/react-devtools-shared/src/devtools/views/ErrorBoundary/ReportNewIssue.js (hunk 1 above)
Comment (bvaughn) [reply packages/react-devtools-shared/src/devtools/views/ErrorBoundary/ReportNewIssue.js:16]: I _think_ I used `encodeURIComponent` initially and it _didn't work_ for some reason so I backed it out 😅  but maybe I should try again to verify.

File: packages/react-devtools-shared/src/devtools/views/ErrorBoundary/ReportNewIssue.js (hunk 2 above)
Comment (bvaughn) [reply packages/react-devtools-shared/src/devtools/views/ErrorBoundary/ReportNewIssue.js:16]: Unclear to me. It didn't replace this string properly before I explicitly escaped it.

**Review by rickhanlonii — APPROVED** [review]
LGTM!

**Comment by sethwebster** [comment]
Love this.

**Pull request merged**